dev/devicelab/lib/framework/metrics_center.dart

```diff
@@ -133,5 +133,43 @@ Future<void> uploadToSkiaPerf(String? resultsPath, String? commitTime, String? t
   resultsJson = json.decode(await resultFile.readAsString()) as Map<String, dynamic>;
   final List<MetricPoint> metricPoints = parse(resultsJson, benchmarkTagsMap, taskName);
   final FlutterDestination metricsDestination = await connectFlutterDestination();
-  await upload(metricsDestination, metricPoints, commitTimeSinceEpoch, taskName);
+  await upload(
+    metricsDestination,
+    metricPoints,
+    commitTimeSinceEpoch,
+    metricFileName(taskName, benchmarkTagsMap),
+  );
+}
+
+/// Create metric file name based on `taskName`, `arch`, `host type`, and `device type`.
+///
+/// Same `taskName` may run on different platforms. Considering host/device tags to
+/// use different metric file names.
+///
+/// This affects only the metric file name which contains metric data, and does not affect
+/// real host/device tags.
+/// 
+/// For example:
+///   Old file name: `backdrop_filter_perf__timeline_summary`
+///   New file name: `backdrop_filter_perf__timeline_summary_intel_linux_motoG4`
+String metricFileName(
+  String taskName,
+  Map<String, dynamic> benchmarkTagsMap,
+) {
+  String fileName = taskName;
```

Comment:
See `StringBuffer`: https://api.dart.dev/dev/2.17.0-270.0.dev/dart-core/StringBuffer-class.html

Reply:
Thanks for the reference! Used `StringBuffer` instead.